code/game/objects/obj_materials.dm

```diff
@@ -0,0 +1,55 @@
+/obj/proc/dismantle()
+	if(!dismantled)
+		dismantled = TRUE
+		create_dismantled_products(get_turf(src))
+		if(!QDELETED(src))
+			qdel(src)
+	. = TRUE
+
+/obj/proc/create_dismantled_products(var/turf/T)
+	if(material)
+		material.place_dismantled_product(T)
+
+/obj/proc/update_from_materials()
+	if(material.products_need_process() && !is_processing)
+		START_PROCESSING(SSobj, src)
+	if(material.conductive)
+		obj_flags |= OBJ_FLAG_CONDUCTIBLE
+	else
+		obj_flags &= (~OBJ_FLAG_CONDUCTIBLE)
+	if(material_alteration & MAT_FLAG_ALTERATION_NAME)
+		update_material_name()
+	if(material_alteration & MAT_FLAG_ALTERATION_DESC)
+		update_material_desc()
+	queue_icon_update()
+
+/obj/proc/update_material_name(var/override_name)
+	var/base_name = override_name || initial(name)
+	if(istype(material))
+		SetName("[material.display_name] [base_name]")
+	else
+		SetName(base_name)
+
+/obj/proc/update_material_desc(var/override_desc)
+	var/base_desc = override_desc || initial(desc)
+	if(istype(material))
+		desc = "[base_desc] It's made of [material.display_name]."
+	else
+		desc = base_desc
+
+/obj/proc/update_material_colour(var/override_colour)
+	var/base_colour = override_colour || initial(color)
+	if(istype(material))
+		color = material.icon_colour
+	else
+		color = base_colour
+
+/obj/on_update_icon()
```

Comment:
I think a /lot/ of on_update_icon() calls don't call parent, I'm like 40% that was the point of making it a separate proc. Maybe it's something that should go into obj update_icon() call so it's reliably called?

Reply:
Well it'd give update_icon() an actual purpose at least I guess.